examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "d8ec9a76-0819-463d-8955-55789ee36400",
+   "metadata": {},
+   "source": [
+    "# Ipyreact walkthrough"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "083f2dc7-c0ec-482f-a720-aa2896ad193e",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Welcome to this ipyreact walkthrough!   \n",
+    "The tutorial will be based on a very simple react button to show all the ipyreact features.  \n",
+    "\n",
+    "**Content** \n",
+    "* use the %react cell magic\n",
+    "* write a widget\n",
+    "* load CSS for your widget\n",
+    "* add parameters to your widgets (traitlets)\n",
+    "* interact with these parameters\n",
+    "* simple traitlet oberservation using `change`\n",
+    "* observe a traitlet and call python function\n",
+    "* observe a traitlet and call JavaScript function\n",
+    "* SOON: call python functions from JavaScript\n",
+    "* loading components from external files\n",
+    "* enable hot reloading  \n",
+    "* enable autocompletion in IDEs\n",
+    "* print a message at class initialization"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "06cb067d-0aea-4851-827d-d07f22466693",
+   "metadata": {},
+   "source": [
+    "First, we will use the **`%react` magic**  from ipyreact.  \n",
+    "The following line registers the cellmagic:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de17b188",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%pip install ipyreact\n",
+    "# This line is for JupyterLite (in case that this takes long, restart the kernel and run this cell again)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d85f911",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext ipyreact"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c70f7253-98d5-4131-bf6b-1ac3433e2985",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%react\n",
+    "\n",
+    "import * as React from \"react\";\n",
+    "\n",
+    "export default function MyButton() {\n",
+    "    return ( < button > X < /button>);\n",
+    "    }"
```

Comment:
```suggestion
    "}"
```

Reply:
I will add all these code changes manually in a notebook, because the "Commit suggestion" button might unintentionally break the notebook format.